Extensions/flashlight/Flashlight.json

```diff
@@ -0,0 +1,115 @@
+{
+  "author": "",
+  "description": "This plugin allows you to switch the flashlight / torch of the device on and off.\n\nWorks on iOS 5+\nWorks on Android 2+\nAndroid 6+ runtime permissions are handled automatically\nWorks on WP8\nDepends on capabilities of the device, so you can test it with an API call",
+  "extensionNamespace": "",
+  "fullName": "FlashLight",
+  "helpPath": "",
+  "iconUrl": "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0ibWRpLWZsYXNobGlnaHQiIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0Ij48cGF0aCBkPSJNOSwxMEw2LDVIMThMMTUsMTBIOU0xOCw0SDZWMkgxOFY0TTksMjJWMTFIMTVWMjJIOU0xMiwxM0ExLDEgMCAwLDAgMTEsMTRBMSwxIDAgMCwwIDEyLDE1QTEsMSAwIDAsMCAxMywxNEExLDEgMCAwLDAgMTIsMTNaIiAvPjwvc3ZnPg==",
+  "name": "FlashLight",
+  "previewIconUrl": "https://resources.gdevelop-app.com/assets/Icons/flashlight.svg",
+  "shortDescription": "This plugin allows you to switch the flashlight / torch of the device on and off.",
+  "version": "1.0.0",
+  "tags": [
+    "flashlight",
+    "android",
+    "light"
+  ],
+  "authorIds": [
+    "jy7FXnGX0ZZcWfrAI9YuQaeIphi1"
+  ],
+  "dependencies": [
+    {
+      "exportName": "cordova-plugin-flashlight",
+      "name": "Flash light",
+      "type": "cordova",
+      "version": "3.2.0"
+    }
+  ],
+  "eventsFunctions": [
+    {
+      "description": "switch the flashlight / torch of the device off ",
+      "fullName": "Switch Off flashlight / torch",
+      "functionType": "Action",
+      "name": "switchOff",
+      "private": false,
+      "sentence": "switch the flashlight / torch of the device off",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "window.plugins.flashlight.available(function(isAvailable) {\n  if (isAvailable) {\n \n   \n     window.plugins.flashlight.switchOn(\n      function() {}, \n      function() {}, \n      {intensity: 0.3} \n    );\n \n  } else {\n    alert(\"Flashlight not found!\");\n  }\n});\n",
```

Comment:
Remove unnecessary extra spaces and lines.
In a JSEvent you can format the code by **right click** and **format**.

Reply:
- You have to check if the flashlight plugin is available.
A desktop game trigger lots of errors, which is normal because `window.plugins.flashlight` and  `window.plugins` don't exist on desktop.

- `alert("Flashlight not found!");` this is a blocking popup, in a game this kind of thing is disturbing. Just remove this alert. If you want trigger an error you can simply use `console.error("my message here")` this kind of error will be visible in the debugger console in GDevelop.